lib/plugins/init/index.js

```diff
@@ -0,0 +1,118 @@
+'use strict';
+
+const BbPromise = require('bluebird');
+const path = require('path');
+
+class Init {
+  constructor(serverless, options) {
+    this.serverless = serverless;
+    this.options = options;
+
+    this.commands = {
+      init: {
+        usage: 'Initializes a provider with the Serverless Framework',
+        lifecycleEvents: [
+          'init',
+        ],
+        options: {
+          provider: {
+            usage: 'Name of the provider',
+            required: true,
+            shortcut: 'p',
+          },
+          key: {
+            usage: 'Access key for the provider',
+            shortcut: 'k',
+          },
+          secret: {
+            usage: 'Secret key for the provider',
+            shortcut: 's',
+          },
+          profile: {
+            usage: 'Name of the profile you wish to create. Defaults to \'default\'',
+            shortcut: 'n',
+          },
+        },
+      },
+    };
+
+    this.hooks = {
+      'init:init': () => BbPromise.bind(this)
+        .then(this.init),
+    };
+  }
+
+  init() {
```

Comment:
The command should be done provider independent so the code here should be in a separate AWS->init plugin that simply uses the main command that is defined